### ISSUE TITLE
Nullability flow analysis ignores dereferences

### ISSUE THREAD

**Issue by sharwell** [dotnet/roslyn#23270]
**Version Used**: 11152017

**Steps to Reproduce**:

```csharp
object? obj = null;
obj.ToString();
obj.ToString();
```

**Expected Behavior**:

One warning is reported (first call to `ToString`).

**Actual Behavior**:

Two warnings are reported. However, there is no way for this code to reach the second `obj.ToString()` if `obj` is null.

**Comment by jcouv** [comment]
Removing the language design label, as the language question was resolved. A dereference `x.y` should be treated by flow analysis like `if (x is null) throw; x.y`.

Notes at https://github.com/dotnet/csharplang/blob/master/meetings/2018/LDM-2018-08-20.md#dereference-of-nullable-types

**Comment by jcouv** [comment]
@cston This is the issue tracking the flow analysis work for dereferences. The design question is whether this would rely on `PreciseAbstractFlowPass._trackExceptions` or some other existing mechanism.

**Comment by jcouv** [comment]
Was fixed by https://github.com/dotnet/roslyn/pull/30564
